docs/source/asr/datasets.rst

```diff
@@ -507,6 +507,127 @@ An example using an AIS cluster at ``hostname:port`` with a tarred dataset for t
+
+* Dynamic batch sizes
+    Lhotse samples mini-batches to satisfy the constraint of total speech duration in a mini-batch (``batch_duration``),
+    rather than a specific number of examples (i.e., batch size).
+* Dynamic bucketing
+    Instead of statically pre-bucketing the data, Lhotse allocates training examples to buckets dynamically.
+    This allows more rapid experimentation with bucketing settings (number of buckets, specific placement of bucket duration bins)
+    to minimize the amount of padding and accelerate training.
+* Quadratic duration penalty
+    Adding a quadratic penalty to an utterance's duration allows to sample mini-batches so that the
+    GPU utilization is more consistent across big batches of short utterances and small batches of long utterances when using
+    models with quadratic time/memory complexity (such as transformer).
+* Dynamic weighted data source multiplexing
+    An approach to combining diverse data sources (e.g. multiple domains, languages, tasks)
+    where each data source is treated as a separate stream with its own sampling probability. The resulting data stream is a
+    multiplexer that samples from each sub-stream. This approach ensures that the distribution of different sources is approximately
+    constant in time (i.e., stationary); in fact, each mini-batch will have roughly the same ratio of data coming from each source.
+    Since the multiplexing is done dynamically, it is very easy to tune the sampling weights.
+
+Lhotse dataloading supports the following types of inputs:
+
+* NeMo manifests
+    Regular NeMo JSON manifests.
+* NeMo tarred data
+    Tarred NeMo JSON manifests + audio tar files; we also support combination of multiple NeMo
+    tarred data sources (e.g., multiple buckets of NeMo data or multiple datasets) via dynamic multiplexing.
+* Lhotse CutSet manifests
+    Regular Lhotse CutSet manifests (typically gzipped JSONL).
+    See `Lhotse Cuts documentation`_ to learn more about Lhotse data formats.
+* Lhotse Shar data
+    Lhotse Shar is a data format that also uses tar files for sequential data loading,
+    but is designed to be modular (i.e., easily extensible with new data sources and with new feature fields).
+    More details can be found here: |tutorial_shar|
+
+.. caution:: As of now, Lhotse is mainly supported in most ASR model configurations. We aim to gradually extend this support to other speech tasks.
+
+.. _Lhotse: https://github.com/lhotse-speech/lhotse
+.. _Lhotse Cuts documentation: https://lhotse.readthedocs.io/en/latest/cuts.html
+.. |tutorial_shar| image:: https://colab.research.google.com/assets/colab-badge.svg
+    :target: https://colab.research.google.com/github/lhotse-speech/lhotse/blob/master/examples/04-lhotse-shar.ipynb
+
+Enabling Lhotse via configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Start with an existing NeMo experiment YAML configuration. Typically, you'll only need to add a few options to enable Lhotse.
```

Comment:
Let's add a comment here that there would be no epoch. Do we ignore the epoch or we need to set it a very large number?

Reply:
I added the following note, hope that clarifies it:

```
.. note:: Using Lhotse with tarred datasets will make the dataloader infinite, 
  ditching the notion of an "epoch". "Epoch" may still be logged in W&B/TensorBoard, 
  but it will correspond to the number of executed training loops between validation loops.
```